java/src/test/java/com/saucelabs/simplesauce/SauceOptionsTest.java

```diff
@@ -1,50 +1,133 @@
 import org.openqa.selenium.edge.EdgeOptions;
 import org.openqa.selenium.firefox.FirefoxOptions;
 import org.openqa.selenium.ie.InternetExplorerOptions;
-import org.openqa.selenium.remote.BrowserType;
 import org.openqa.selenium.safari.SafariOptions;
 
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 
 import static org.junit.Assert.assertEquals;
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.spy;
 import static org.openqa.selenium.UnexpectedAlertBehaviour.DISMISS;
 
-public class SauceOptionsTest extends BaseConfigurationTest{
+public class SauceOptionsTest {
+    protected SauceOptions sauceOptions = new SauceOptions();
+
+    @Rule
+    public MockitoRule initRule = MockitoJUnit.rule();
+
     @Test
-    public void usesLatestChromeWindowsVersions() {
-        assertEquals(BrowserType.CHROME, sauceOptions.getBrowserName());
-        assertEquals("Windows 10", sauceOptions.getPlatformName());
+    public void usesLatestChromeWindowsVersionsByDefault() {
+        assertEquals(Browser.CHROME, sauceOptions.getBrowserName());
+        assertEquals(SaucePlatform.WINDOWS_10, sauceOptions.getPlatformName());
         assertEquals("latest", sauceOptions.getBrowserVersion());
     }
 
     @Test
     public void updatesBrowserBrowserVersionPlatformVersionValues() {
-        sauceOptions.setBrowserName(BrowserType.FIREFOX);
-        sauceOptions.setPlatformName(Platforms.MAC_OS_HIGH_SIERRA.getOsVersion());
+        sauceOptions.setBrowserName(Browser.FIREFOX);
+        sauceOptions.setPlatformName(SaucePlatform.MAC_HIGH_SIERRA);
         sauceOptions.setBrowserVersion("68");
 
-        assertEquals(BrowserType.FIREFOX, sauceOptions.getBrowserName());
+        assertEquals(Browser.FIREFOX, sauceOptions.getBrowserName());
+        assertEquals("68", sauceOptions.getBrowserVersion());
+        assertEquals(SaucePlatform.MAC_HIGH_SIERRA, sauceOptions.getPlatformName());
+    }
+
+    @Test
+    public void fluentPatternWorks() {
+        sauceOptions.setBrowserName(Browser.FIREFOX)
+                .setBrowserVersion("68")
+                .setPlatformName(SaucePlatform.MAC_HIGH_SIERRA);
+
+        assertEquals(Browser.FIREFOX, sauceOptions.getBrowserName());
         assertEquals("68", sauceOptions.getBrowserVersion());
-        assertEquals("macOS 10.13", sauceOptions.getPlatformName());
+        assertEquals(SaucePlatform.MAC_HIGH_SIERRA, sauceOptions.getPlatformName());
     }
 
     @Test
-    @Ignore("Not Implemented Yet")
     public void acceptsOtherW3CValues() {
+        sauceOptions.setPageLoadStrategy(PageLoadStrategy.EAGER);
+        sauceOptions.setAcceptInsecureCerts(true);
+        Map<Timeouts, Integer> timeouts = new HashMap<>();
+        timeouts.put(Timeouts.IMPLICIT, 1);
+        timeouts.put(Timeouts.PAGE_LOAD, 100);
+        timeouts.put(Timeouts.SCRIPT, 10);
+        sauceOptions.setTimeouts(timeouts);
```

Comment:
So will this still be possible? It's not getting replaced with the new impl that you supplied?
https://github.com/saucelabs/simple_sauce/pull/115#discussion_r363121483

Reply:
I made a bunch of test updates yesterday (I'm going to wait for a new PR) that changed these lines to the new method.

That being said, the old method still has to be supported so that we can deserialize from a config.